zerver/views/message_send.py

```diff
@@ -339,3 +348,77 @@ def render_message_backend(
 
     rendering_result = render_markdown(message, content, realm=user_profile.realm)
     return json_success(request, data={"rendered": rendering_result.rendered_content})
+
+
+REPORT_MESSAGE_REASONS = {
+    "spam": f"{_('Spam')}",
+    "harassment": f"{_('Harassment')}",
+    "inappropriate": f"{_('Inappropriate content')}",
+    "norms": f"{_('Violates community norms')}",
+    "other": f"{_('Other reason')}",
+}
+
+# must be kept in sync with ./static/shared/js/fenced_code.js
+def get_unused_fence(content: str) -> str:
+    # in python, it's one line of code get a uniquely long fence
+    return "`" * (1 + max([2] + [len(x) for x in re.findall(r"`+", content)]))
+
+
+def get_quoted_text(content: str) -> str:
+    fence = get_unused_fence(content)
+    return f"\n{fence}quote\n{content}\n{fence}\n"
+
+
+@has_request_variables
+def report_message_backend(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    message_id: int = REQ(converter=to_non_negative_int, path_only=True),
+    reason: str = REQ(
+        default="other", str_validator=check_string_in(list(REPORT_MESSAGE_REASONS.keys()))
+    ),
+    explanation: Optional[str] = REQ(default=None),
```

Comment:
Might as well use `explanation: str = REQ(default="")` if we’re going to convert to `""` anyway.

Reply:
DONE. good catch.